### PR TITLE
CI: Add formatting checks and migrate to Github Actions

### PR DESCRIPTION
**Pull request by josephlr**
We use both `gofmt` and `goimports` so that we get the benefits of
`gofmt -s` while also making sure our imports are clean.

The simulator code is changed so that now you can build and test on windows.

The Cirrus configuration is deleted, and a Github Actions configuration is used instead. 

I wasn't quite able to get everything working on `windows-latest` (still running into linker issues), but the exact same instructions worked on my desktop system, so I have no idea what's going on.

### CODE REVIEW

**Comment by josephlr** [comment]
@jkl73 @alexmwu this is ready for review

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ RUN apt-get update && apt-get install -y \
     libssl-dev libtspi-dev protobuf-compiler \
  && rm -rf /var/lib/apt/lists/*
 # We need golangci-lint for linting
-ARG VERSION=1.41.1
+ARG VERSION=1.42.0
```
Comment (alexmwu) [inline Dockerfile:7]: Can we remove this file altogether?

File: simulator/internal/internal.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ package internal
 // // Store NVDATA in memory, and we don't care about updates to failedTries.
 // #cgo CFLAGS: -DVTPM=NO -DSIMULATION=NO -DUSE_DA_USED=NO
 // // Flags from ../ms-tpm-20-ref/TPMCmd/configure.ac
-// #cgo CFLAGS: -std=gnu11 -Wall -Wformat-security -fstack-protector-all -fPIC
+// #cgo CFLAGS: -std=gnu11 -Wall -Wformat-security -fPIC
```
Comment (alexmwu) [inline simulator/internal/internal.go:17]: jw: why won't this work on windows or macOS?

File: README.md [hunk 3]
```diff
@@ -103,7 +117,7 @@ Remember to remove the lines from `simulator/internal/internal.go` before
 committing changes.
 ```
 // #cgo CFLAGS: -I $OPENSSL_PATH/include
-// #cgo LDFLAGS: -L$OPENSSL_PATH/lib
+// #cgo LDFLAGS: -L $OPENSSL_PATH/lib
```
Comment (alexmwu) [inline README.md:120]: maybe tell the user to replace existing lines prefixed with "#cgo darwin CFLAGS" now that we've added the lines in for Github actions

```
// #cgo darwin CFLAGS: -I $OPENSSL_PATH/include
// #cgo darwin LDFLAGS: -L $OPENSSL_PATH/lib
```

**Review by alexmwu — APPROVED** [review]
remove the "CI: Add formatting checks" and "CI: update golangci-lint to 1.42.0" commits; they're deleted afterwards anyway.

File: README.md (hunk 3 above)
Comment (josephlr) [reply README.md]: Updated the `README`

File: simulator/internal/internal.go (hunk 2 above)
Comment (josephlr) [reply simulator/internal/internal.go:17]: This flag actually works on macOS and Linux. I changed the code to just only have the flag present on `!windows`.

File: Dockerfile (hunk 1 above)
Comment (josephlr) [reply Dockerfile]: Removed

**Pull request merged**